website/src/blogs/mayastor-nvme-of-tcp-performance.md

```diff
@@ -0,0 +1,183 @@
+Before we get started, you might wonder how we came to this point that a new project like OpenEBS Mayastor was able to deliver amongst the fastest storage available today. Richard Elling of Viking / Sanmina recently wrote an excellent blog on the trends in hardware design that puts NVMe and OpenEBS Mayastor into context:  [https://richardelling.substack.com/p/the-pendulum-swings-hard-towards](https://richardelling.substack.com/p/the-pendulum-swings-hard-towards)
+
+## Hardware setup
+
+Let’s get to it.  We will be using three machines which will run kernel version 5.8. The hardware configuration of each host is as follows:
+
+- Intel(R) Xeon(R) Gold 6252 CPU @ 2.10GHz
+- Intel Corporation NVMe Datacenter SSD [Optane]
+- Mellanox Technologies MT28908 Family [ConnectX-6]
+
+## Baseline performance
+
+To understand the performance of the device we will be using throughout the test, we run the following Fio workload:
+
+    [global]
+    ioengine=linuxaio
+    thread=1
+    group_reporting=1
+    direct=1
+    norandommap=1
+    bs=4k
+    numjobs=8
+    time_based=1
+    ramp_time=0
+    runtime=300
+    iodepth=64
+    
+    
+    [random-read-QD64]
+    filename=/dev/nvme1n1
+    rw=randread
+    stonewall
+    
+    
+    [random-write-QD64]
+    filename=/dev/nvme1n1
+    rw=randwrite
+    stonewall
+    
+    
+    [random-rw-QD64]
+    filename=/dev/nvme1n1
+    rw=randrw
+    stonewall
+
+![](/images/blog/mayastor-nvme1.png)
+These numbers are incredibly high and are provided by a ***single*** device. Note that the benchmark itself is rather synthetic in the sense that, in practice, no workload is 100% random.
+
+## High-level overview of the experiments
+
+My approach in this benchmarking is very much OpenEBS Mayastor “the hard way”.  If you want an easier to use solution that for example automates pool creation and device selection and so on - we call that offering Kubera Propel (also open source btw). You can learn more about Kubera Propel on the [MayaData.io](https://mayadata.io/) web site.  
+
+On two of the nodes, we create a pool (MSP CRD) which we use in the control plane to determine replica placement. To construct pools, we must have what we call a persistence layer. We support several ways to access this persistence layer. To select a particular scheme we use URIs. In this case we will be using today the pcie:/// scheme. This means that Mayastor will directly write into the NVMe devices listed above. The nice thing is that from the user perspective, things do not change that much. We simply replace disks: [‘/dev/nvme0n1’] with disks: [‘pcie:///80:00.0’]. Note that this persistence layer is used to store the replicas of the PVC. Once we have this layer up and running, we will create storage classes and select that we want to use nvmf (NVMe-oF) as the transport layer between the replicas, resulting in NVMe all the way through. 
+
+After we have deployed mayastor we applied to following two storage classes:
+
+    kind: StorageClass
```

Comment:
how is this working (converting into code block) since we haven't put any backtick? 🤔

Reply:
1 tab or 4 spaces for pre-formatted code

Reply:
I believe this is answered already.